Address PR comments Part 2

issackjohn opened the pull request:
![image](https://github.com/lpardosixtosMs/Speedometer/assets/117320405/6086e30d-a2a8-4d67-be76-c54c3f970ffa)
Hosted version: https://issackjohn.github.io/Speedometer/resources/todomvc/big-dom-generator/dist/

issackjohn committed:
combine folderwrapper and treeitem

issackjohn committed:
update tree-generator logic

issackjohn committed:
forgot to remove list

issackjohn committed:
reduce visible depth

issackjohn committed:
update dists

issackjohn commented:
@lpardosixtosMs I had to reduce the `export const MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH = 9;` because it looked bad visually, this is something we can review.

issackjohn committed:
update comment

lpardosixtosMs commented on resources/todomvc/big-dom-generator/src/components/tree-area.jsx:
```diff
@@ -32,29 +20,25 @@ const TreeItem = (props) => {
                 <span className="just-span spectrum-TreeView-itemLabel">{isExpandableItem ? "Sprint" : "Todo List"}</span>
             </a>
-            {children}
+            {children && <ul className="spectrum-TreeView spectrum-TreeView--sizeS">{children}</ul>}
```
Use `isExpandableItem` instead.

lpardosixtosMs commented on resources/todomvc/big-dom-generator/src/tree-generator.js:
```diff
@@ -4,44 +4,39 @@ import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_
+ * @param {Object} options - The options object.
+ * @param {number} options.expandableItemWeight - The weight for the "expandableItem" node type. <li></li> with ChevronRight svg.
+ * @param {number} options.nonExpandableItemWeight - The weight for the "nonExpandableItem" node type. <li></li> TaskListIcon svg.
  * @returns {Object} The generated tree structure. Example structure:
```
Specify which are the values for `type`

issackjohn replied:
added the comment, let me know if I should reword it.

lpardosixtosMs commented on resources/todomvc/big-dom-generator/src/tree-generator.js:
```diff
@@ -64,18 +59,6 @@ export const generateTreeHead = ({ listWeight, expandableItemWeight, nonExpandab
             let currentNode = treeNodes[index];
             switch (currentNode.type) {
                 case "expandableItem":
```
Use an if instead.

lpardosixtosMs commented on resources/todomvc/big-dom-generator/src/tree-generator.js:
```diff
@@ -86,6 +69,15 @@ export const generateTreeHead = ({ listWeight, expandableItemWeight, nonExpandab
+                case "nonExpandableItem":
+                    if (random.coin(PROBABILITY_OF_HAVING_CHILDREN)) {
+                        currentNode.type = "expandableItem";
+                        currentNode.children = [{ type: "expandableItem", children: [] }];
```
Randomly decide if the child will be expandable or nonExpandable.  Make sure you modify the weigtht too.

issackjohn replied:
@lpardosixtosMs I recall there was also a comment you wanted me to add here?

issackjohn committed:
address comments.

issackjohn committed:
build static html file for big-dom-generator

issackjohn committed:
rebuild all dists

issackjohn committed:
8 is the golden number for visible tree item depth

issackjohn committed:
rebuild all complex dist

issackjohn commented:
Updated hosted version and also changed the visible tree item depth to 8. PTAL.
![image](https://github.com/lpardosixtosMs/Speedometer/assets/117320405/e429b0c6-79bb-4286-9a21-bb262b528380)

issackjohn commented:
![image](https://github.com/lpardosixtosMs/Speedometer/assets/117320405/5af11c15-d0ba-4c54-8085-f79a381191fd)

issackjohn committed:
reword the comment

issackjohn committed:
better wording part 2

issackjohn committed:
should not use childType here

issackjohn committed:
rebuild all dists

lpardosixtosMs commented on resources/todomvc/big-dom-generator/src/tree-generator.js:
```diff
@@ -62,32 +58,30 @@ export const generateTreeHead = ({ listWeight, expandableItemWeight, nonExpandab
+                    totalNodes += nodeWeight[childType];
+                    treeNodes.push(currentNode.children[0]);
+                }
+            } else {
```
When I suggested turning this into an if, I thought it would be just if ... else, without the last check. I suggest to either remove the last else or turn it back into a switch, whichever you prefer.

issackjohn replied:
I like having the error check there so I'll turn it back to a switch.

issackjohn committed:
turn if-else back into a switch statement
